src/cbc_sdk/platform/devices.py

```diff
@@ -334,8 +331,14 @@ def get_asset_group_ids(self, membership="ALL"):
         Returns:
             list[str]: A list of asset group IDs this device belongs to.
         """
-        rc = Device.get_asset_groups_for_devices(self._cb, self, membership)
-        return rc[self._model_unique_id]
+        if membership not in Device.VALID_ASSETGROUP_FILTERS:
+            raise ApiError(f"Invalid filter value: {membership}")
+        if membership == "ALL":
+            return [g['id'] for g in self._info['asset_group']]
+        elif membership == "MANUAL":
```

Comment:
wondering whether it makes sense for last two to be combined `return [g['id'] for g in self._info['asset_group'] if g['membership_type'] == membership]`

Reply:
Ideally, I'd like to make sure the input value "MANUAL" and the value "MANUAL" we feed to the server are considered as "disjoint." That way, if the underlying API ever changes, the values the SDK accepts need not.